File: 1371.py
class Solution:
    def check(self,x):
        if x in {'a','e','i','o','u'}:
            return True
        else:
            return False
    def findTheLongestSubstring(self, s: str) -> int:
        mp=dict()
        pre=0
        result=0
        for p,i in enumerate(s):
            if self.check(i):
                pre=pre^ord(i)
            if pre==0:
                result=p+1
            else:
                if pre not in mp:
                    mp[pre]=p
                else:
                    temp=p-mp[pre]
                    if temp>result:
                        result=temp
        
        return result

File: 1371.c
class Solution {
public:
    bool check(char x)
    {
        if(x=='a'||x=='e'||x=='i'||x=='o'||x=='u')
            return true;
        else
            return false;
    }
    int findTheLongestSubstring(string s) {
        int n=s.length();
        int result=0;
        unordered_map<int,int> map;
        int pre=0;
        for(int i=0;i<n;i++)
        {
            if(check(s[i]))
            {
                pre=pre^int(s[i]);
            }
            if(pre==0)result=i+1;
            else{
                auto got=map.find(pre);
                if(got==map.end())
                    map.insert({pre,i});
                else{
                    result=i-got->second>result?i-got->second:result;
                }
            }
        }
        return result;
    }
};